pkg/runtime/wfengine/component.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	// Start time is also optional and must be in the RFC3339 format (e.g. 2009-11-10T23:00:00Z).
+	if req.Options != nil {
+		if startTimeRFC3339, ok := req.Options["dapr.workflow.start_time"]; ok {
+			if startTime, err := time.Parse(time.RFC3339, startTimeRFC3339); err != nil {
+				return nil, fmt.Errorf("start times must be in RFC3339 format (e.g. \"2009-11-10T23:00:00Z\")")
+			} else {
+				opts = append(opts, api.WithStartTime(startTime))
+			}
+		}
+	}
+
+	var workflowID api.InstanceID
+	var err error
+
+	workflowID, err = c.client.ScheduleNewOrchestration(ctx, req.WorkflowName, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to start workflow: %w", err)
+	}
+
+	c.logger.Infof("created new workflow instance with ID '%s'", workflowID)
+	wfRef := &workflows.WorkflowReference{
+		InstanceID: string(workflowID),
+	}
+	return wfRef, nil
+}
+
+func (c *workflowEngineComponent) Terminate(ctx context.Context, req *workflows.WorkflowReference) error {
+	if req.InstanceID == "" {
+		return fmt.Errorf("a workflow instance ID is required")
+	}
+
+	if err := c.client.TerminateOrchestration(ctx, api.InstanceID(req.InstanceID), ""); err != nil {
+		return fmt.Errorf("failed to terminate workflow %s: %w", req.InstanceID, err)
+	}
+
+	c.logger.Infof("scheduled termination for workflow instance '%s'", req.InstanceID)
+	return nil
+}
+
+func (c *workflowEngineComponent) Get(ctx context.Context, req *workflows.WorkflowReference) (*workflows.StateResponse, error) {
+	if req.InstanceID == "" {
+		return nil, fmt.Errorf("a workflow instance ID is required")
+	}
+
+	if metadata, err := c.client.FetchOrchestrationMetadata(ctx, api.InstanceID(req.InstanceID)); err != nil {
+		return nil, fmt.Errorf("failed to get workflow metadata for '%s': %w", req.InstanceID, err)
+	} else {
+		res := &workflows.StateResponse{
+			WFInfo: workflows.WorkflowReference{
+				InstanceID: req.InstanceID,
+			},
+			StartTime: metadata.CreatedAt.Format(time.RFC3339),
+			Metadata: map[string]string{
+				"dapr.workflow.name":           metadata.Name,
+				"dapr.workflow.runtime_status": getStatusString(int32(metadata.RuntimeStatus)),
+				"dapr.workflow.input":          metadata.SerializedInput,
+				"dapr.workflow.custom_status":  metadata.SerializedCustomStatus,
+				"dapr.workflow.last_updated":   metadata.LastUpdatedAt.Format(time.RFC3339),
+			},
+		}
+
+		// Status-specific fields
+		if metadata.FailureDetails != nil {
+			res.Metadata["dapr.workflow.failure.error_type"] = metadata.FailureDetails.ErrorType
+			res.Metadata["dapr.workflow.failure.error_message"] = metadata.FailureDetails.ErrorMessage
+		} else if metadata.IsComplete() {
+			res.Metadata["dapr.workflow.output"] = metadata.SerializedOutput
+		}
+
+		return res, nil
+	}
+}
+
+// Status values are defined at: https://github.com/microsoft/durabletask-go/blob/119b361079c45e368f83b223888d56a436ac59b9/internal/protos/orchestrator_service.pb.go#L42-L64
+func getStatusString(status int32) string {
```

Comment:
pure nit: use maps instead

```golang
var statusMap = map[int32]string {...}

if statusStr, ok := statusMap[status]; ok {
    return statusStr
}
return "UNKNOWN"
```

Reply:
I figured this would be more efficient, but if maps are preferred over switch in cases like this then I'm happy to do it.